whatsapp_bot.py

```diff
@@ -0,0 +1,164 @@
+        receiver_contact:
+            Receiver's contact (ie, my contact), follows a format 'whatsapp:+<phone number>'
+
+    Returns:
+        Response text
+    """
+    text_message = incoming_message.lower()
+    # 'explain' response
+    if text_message[0:7] == "explain":
+        response = ActivityGenerator.generate(
+            model=LLM,
+            temperature=TEMPERATURE,
+            messages=[ELI3_MESSAGES],
+            message_kwargs={"input": text_message[7:].strip()},
+        )
+        return response["choices"][0]["message"]["content"]
+    # 'activities' response
+    elif "activities" in text_message[0:10]:
+        EYFS_PARAMETERS["description"] = text_message
+        thread = Thread(target=send_text, args=[EYFS_MESSAGES, EYFS_PARAMETERS, receiver_contact, sender_contact])
+        thread.start()
+        return "Thank you for your question. I am thinking..."
+    else:
+        # Return a default message
+        return 'Write "Explain <your question>" to explain a concept to a 3-year old \n\n or" \
+             + "\n\n "Activities <your topic>" to get activity ideas'
+
+
+def send_text(messages: List[Dict], message_kwargs: Dict, my_contact: str, receiver_contact: str) -> None:
+    """Generate text messages and send them to a given contact
+
+    Args:
+        messages:
+            List of messages to be used as prompts
+        message_kwargs:
+            Dictionary of keyword arguments to be passed to the ActivityGenerator
+        my_contact:
+            Sender's contact, follows a format 'whatsapp:+<phone number>'
+        receiver_contact:
+            Receiver's contact (ie, my contact), follows a format 'whatsapp:+<phone number>'
+    """
+    # Generate response to the message
+    response = ActivityGenerator.generate(
+        model=LLM,
+        temperature=TEMPERATURE,
+        messages=messages,
+        message_kwargs=message_kwargs,
+    )
+    text_body = response["choices"][0]["message"]["content"]
+    # Format the text_body for better display on WhatsApp
+    text_body = format_activities_text(text_body)
+    # Divide output into 1500 character chunks due to WhatsApp character limit of 1600 chars
+    texts = [text_body[i : i + 1500] for i in range(0, len(text_body), 1500)]
+    # Send message
+    for text in texts:
+        client.messages.create(body=text, from_=my_contact, to=receiver_contact)
+        sleep(0.5)
+    return
+
+
+@app.route("/")
+def hello() -> str:
+    """Test the api is working"""
+    return "Hello, World!"
+
+
+@app.route("/text", methods=["POST"])
+def text_reply() -> str:
+    """Respond to incoming messages"""
+    reply = generate_reply(
+        incoming_message=request.form.get("Body"),
+        sender_contact=request.form.get("From"),
+        receiver_contact=request.form.get("To"),
+    )
+    resp = MessagingResponse()
+    resp.message(reply)
+    return str(resp)
+
+
+if __name__ == "__main__":
+    app.run(debug=True)
```

Comment:
We should remove the `debug=True` flag before merging this PR

Reply:
Will do!